opm/simulators/flow/FlowMainEbos.hpp

```diff
@@ -219,6 +219,18 @@ namespace Opm
             std::cout << "*          For more information, see https://opm-project.org         *\n";
             std::cout << "*                                                                    *\n";
             std::cout << "**********************************************************************\n\n";
+
+#ifdef _OPENMP
+            const int threads = 1;
+#else
+            const int threads = omp_get_max_threads();
+#endif
```

Comment:
Am I reading this wrong?  This appears to invoke `omp_get_max_threads()` when `_OPENMP` is **not** defined.

Reply:
I agree that looks flipped, another thing: I do not think we use the maximum number of threads, but instead default to 2 threads, if OPENMP is present or 1 if not. On my OpemMP-less build
```
bin/flow --help | grep threads
```
returns
```
    --threads-per-process=INTEGER                 The maximum number of threads to be instantiated per process ('-1' means 'automatic'). Default: 1
```

Reply:
omp_get_max_threads() returns the number of threads that will be used in the next parallel section without a num_threads clause. it isnot the amount of hardware threads. ie, this is correct.

Reply:
Thanks for the explanation! BTW, what should one use to get the amount of hardware threads then? Also: I assume the call to set the number of threads to be used has been made before the call site?

Reply:
omp_get_num_procs() will tell you the number of physical/ logical cores. but you really should not use this because OMP_NUM_THREADS env var is the authority.

Reply:
The number of threads used in any serial section is `1`, that is what `opm_get_num_thread()` returns. `opm_get_max_threads` returns the maximum available. That is the value that we set (default is 2 if there are at least 2 processors).